pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -179,6 +179,12 @@ func (p *tinkerbellProvider) SetupAndValidateCreateCluster(ctx context.Context,
 		return fmt.Errorf("failed setup and validations: %v", err)
 	}
 
+	// ValidateHardwareConfig performs a lazy load of hardware configuration. Given subsequent steps need the hardware
```

Comment:
That's a good catch 😄  
Will do a follow up PR to decouple this.

Reply:
I have some ideas around how you can do this that would make dependencies more explicit for each validation method if you'd like to chat.

Reply:
Definitely! If you already started on it feel free to raise a PR with what you had in mind 👍

Reply:
Working on it. I found some other oddities so I might propose something a little more elaborate.